src/us/kbase/workspace/database/provenance/EventDate.java

```diff
@@ -0,0 +1,119 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Objects;
+import us.kbase.workspace.database.Util;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+import java.time.Year;
+import java.time.YearMonth;
+
+/**
+ * The date of a specified event.
+ */
+public class EventDate {
+
+	private final String date;
+	private final Event event;
+
+	private EventDate(
+			final String date,
+			final Event event) {
+		this.date = date;
+		this.event = event;
+	}
+
+	/**
+	 * Gets the date, for example "2022-05-10".
+	 * Dates are returned in the format yyyy-MM-dd; if no data is available for the
+	 * month or day, the string is truncated to yyyy-MM or yyyy respectively.
+	 *
+	 * @return the date.
+	 */
+	public String getDate() {
+		return date;
+	}
+
+	/**
+	 * Gets the event that occurred on the date in question, for example {@link Event#UPDATED}.
+	 *
+	 * @return the event.
+	 */
+	public Event getEvent() {
+		return event;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(date, event);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		EventDate other = (EventDate) obj;
+		return Objects.equals(date, other.date) && Objects.equals(event, other.event);
+	}
+
+	/**
+	 * Builds an {@link EventDate}.
+	 *
+	 * @param date  the date when the event occurred,
+	 *              in the format yyyy-MM-dd, yyyy-MM, or yyyy.
+	 * @param event the event that occurred on that date, as a string
+	 * @return the new EventDate.
```

Comment:
```suggestion
	 * @return the new {@link EventDate}.
```
Seems trivial but the main benefit is that auto refactoring can change the name here if the class name is refactored